gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/TaskDtoBuilder.java

```diff
@@ -42,50 +41,70 @@ public TaskDto getTaskDto(OffsetDateTime timestamp) {
     }
 
     public List<TaskDto> getListTasksDto(LocalDate businessDate) {
-        List<TaskDto> listTasks = new ArrayList<>();
         ZoneId zone = ZoneId.of(properties.getProcess().getTimezone());
         LocalDateTime businessDateStartTime = businessDate.atTime(0, 30);
+        LocalDateTime businessDateEndTime = businessDate.atTime(23, 30);
         ZoneOffset zoneOffSet = zone.getRules().getOffset(businessDateStartTime);
-        OffsetDateTime timestamp = businessDateStartTime.atOffset(zoneOffSet);
-        while (timestamp.getDayOfMonth() == businessDate.getDayOfMonth()) {
-            listTasks.add(getTaskDto(timestamp));
-            timestamp = timestamp.plusHours(1).atZoneSameInstant(zone).toOffsetDateTime();
+        OffsetDateTime startTimestamp = businessDateStartTime.atOffset(zoneOffSet);
+        OffsetDateTime endTimestamp = businessDateEndTime.atOffset(zoneOffSet);
+        Set<Task> tasks = taskRepository.findAllByTimestampBetweenForBusinessDayView(startTimestamp, endTimestamp);
+        Map<OffsetDateTime, TaskDto> taskMap = new HashMap<>();
+        for (OffsetDateTime loopTimestamp = startTimestamp;
+             !loopTimestamp.isAfter(endTimestamp);
```

Comment:
After a good night of sleep, I think we could perhaps change this condition with `endTimestamp.isBefore(loopTimestamp)` which does absolutely the same as `!loopTimestamp.isAfter(endTimestamp)` but gets rid of the negation that may easily be forgotten.

That's only for readability purpose, this one should already work fine.

Reply:
which misses the case equals enddate ...

Reply:
You're right, I forgot the case where loopTimestamp is equal to endTimestamp, which should also be treated in the loop.
Indeed we can't get rid of the negation here as there is no `isBeforeOrEqual` method.